### ISSUE TITLE
NYSIIS encoding of 'CHRISTINA' 

### ISSUE THREAD

**Issue by ahood** [howardjp/phonics#13]
Noticed phonics::nysiis('CHRISTINA') outputs 'CHRASTAN' (for maxCodeLen >= 8) whereas it should be 'CRASTAN' as per original algorithm (see https://naldc.nal.usda.gov/download/27833/PDF or https://www.springer.com/us/book/9780387695020 and the somewhat more vague https://en.wikipedia.org/wiki/New_York_State_Identification_and_Intelligence_System; can't find original report by Taft).

 Looks like discrepancy is due to the omission of the first letter of the name in nysiis.R line 107, i.e.
`word <- substr(word, 2, nchar(word))` before the application of the 'H' rule ([Step 4.5](https://en.wikipedia.org/wiki/New_York_State_Identification_and_Intelligence_System)).

**Comment by howardjp** [comment]
@ahood, thank you for posting this!  I think there may be a discrepancy in expectations.  There are two different algorithms that use the name NYSIIS and we use the "modified" flag and "Christina" is one of the different cases.  Observe:

    > nysiis("CHRISTINA", maxCodeLen = 10)
    [1] "CHRASTAN"
    > nysiis("CHRISTINA", maxCodeLen = 10, modified = TRUE)
    [1] "CRASTAN"

You will find that different algorithms using the name nysiis may produce either output, so this is our solution.

**Comment by ahood** [comment]
What source did you use for your implementation of nysiis_original()? 

Asking because the original source by Taft is hard to come by, and the steps written on Wikipedia (which may or may not be the same as Taft's) are awfully vague to the point where I've seen three different interpretations by three different implementers. The steps in Appendix B of [this USDA report](https://naldc.nal.usda.gov/download/27833/PDF) are much less ambiguous (although I don't know if they are different from Taft's), and include both an "original" and "modified" version; the "original" version is the one I used to arrive at "CRASTAN" as the encoding for Christina.

**Comment by ahood** [comment]
@chrislit Yes, agreed, I don't see where the idea of chopping off the first letter before processing the rest of the word and then gluing it back on came from, and it leads to contradictions, as you have pointed out.

**Comment by howardjp** [comment]
Okay, I am convinced.  I'll get a patch together.  Can you send a few test cases I can add to the test white to exercise it?

**Comment by ahood** [comment]
Just created a pull request with test cases and modifications to nysiis_original() (didn't touch nysiis_modified()). Verified that corrected and new test cases agree with output of abydos.phonetic.nysiis(), authored by @chrislit who also seemed to be working from the original source.

**Comment by howardjp** [comment]
Good question.  CRAN is closed until after the New Year.  I will put together the update in a day or so, but there's no rush since it won't update until January, regardless.

BTW, you seem to be pushing NYSIIS quite hard.  Is there anything I can do to assist?

**Comment by ahood** [comment]
Great, just wondering when I should look out for it and update the packages at work. 

 I was translating some patient-matching code from R to Python, where some of the features involve comparisons between NYSIIS encodings, and I noticed that phonics::nysiis() and jellyfish.nysiis() gave me different results on some of my test cases. So then I saw those vague rules on Wikipedia that everyone was interpreting differently and all the different implementations and I just couldn't let it go :)

**Comment by howardjp** [comment]
Oh, yeah, I know the feeling!  Since CRAN reopens today, I will be
submitting this at some point today.  I have no idea what kind of backlog
on submissions they are expecting, so maybe 24-48 hours from now?

Cool, let me know if I can help any further!

**Comment by howardjp** [comment]
@ahood, phonics 1.2.3 has been approved for distribution and is going to CRAN now
